File: pset6/mario/more/mario.py
from cs50 import get_int

# Prompt user for a pyramid's height
while True:
    height = get_int("Height: ")
    if height > 0 and height < 9:
        break

# Loop over rows
for row in range(height):
    spaces = height - (row + 1)
    hashes = height - spaces
    print(" " * spaces, "#" * hashes, " " * 2, "#" * hashes, sep="")

File: pset1/mario/less/mario.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    // Take user input
    int height;
    do
    {
        height = get_int("Height: ");
    }
    while (height < 0 || height > 23);

    // Build a half-pyramid
    int hashesStartAt = height - 1;
    while (hashesStartAt >= 0)
    {
        // Draw a row
        for (int column = 0; column <= height; column ++)
        {
            if (column < hashesStartAt)
            {
                printf(" ");
            }
            else
            {
                printf("#");
            }
        }
        printf("\n");
        hashesStartAt --;
    }


    // Optional solution looping over rows explicitly
    // for (int row = height; row > 0; row --)
    // {
    //     // Draw columns
    //     for (int column = 0; column <= height; column ++)
    //     {
    //         if (column < hashesStartAt)
    //         {
    //             printf(" ");
    //         }
    //         else
    //         {
    //             printf("#");
    //         }
    //     }
    //     printf("\n");
    //     hashesStartAt --;
    // }
}